### ISSUE TITLE
Fix `font-family-no-missing-generic-family-keyword` false positives for `math`

### ISSUE THREAD

**Issue by otomad** [stylelint/stylelint#8436]
### What minimal example or steps are needed to reproduce the bug?

```css
math {
	font-family: "Cambria Math", math;
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
    "rules": {
        "font-family-no-missing-generic-family-keyword": true
    }
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIC2BDALgCwATADoDtc0AzCfNAWmJSQEsAbAT3i1xAGEaAjAJ1pSwBZdBjYAaLKkwBuAgF8CIMSGIMYAORpxEUjADowAZ0NLwZVQHMEIPPiysQMAB5oY+KIbYsA2gXv22QzRGehh6WnIKSHxLCiCUdxQeKDY-LABdeVNoywAxCB4pawArQzJTWAAHE0RbAJAgkLCItC8HenQYIPE0wODQ8Mic2gs4tASoJJSEds7ukCy5IA)

### Which Stylelint-related dependencies are you using?

```json
{
    "stylelint": "16.15.0"
}
```

### What did you expect to happen?

Stylelint [16.15.0](https://github.com/stylelint/stylelint/releases/tag/16.15.0) #8395 fixed #8390 `font-family-no-missing-generic-family-keyword` false positives for `math`, `emoji` and `fangsong`.

In fact, only `emoji` and `fangsong` were available, but `math` was still unavailable.

Note that if font-family contains only `math` itself, it will pass, but if it also contains other fonts, it will still report an error.

### What actually happened?

The following problems were reported:

```
math.css
 2:15   ✖  Unexpected missing generic font family (font-family-no-missing-generic-family-keyword)
```

**Comment by Mouvedia** [comment]
I didn't look into it but I am pretty sure it's because that `math` is part of 
https://github.com/stylelint/stylelint/blob/cc94919d2b5bf6e030895c026a837c5c9f0d754c/lib/reference/keywords.mjs#L14
and
https://github.com/stylelint/stylelint/blob/cc94919d2b5bf6e030895c026a837c5c9f0d754c/lib/reference/keywords.mjs#L122

**Comment by romainmenke** [comment]
Indeed, the issue is here: https://github.com/stylelint/stylelint/blob/cc94919d2b5bf6e030895c026a837c5c9f0d754c/lib/utils/findFontFamily.mjs#L78-L81

The `findFontFamily` is too simplistic, it scans values looking for things that could be font-families and are unlikely to be something else. It looks mostly at the types and values themselves. This is incorrect. The position/order should also be taken into account.

I think we should split this up so that non-shorthand checks use a much simpler function. `font-face: "Latin Modern Math", math` isn't a shorthand and doesn't require complex parsing to list the font family names.

I wonder if we can use `csstree` here?

This interface seems to imply that `csstree` does give some feedback on the parsed values:

https://csstree.github.io/docs/syntax/#Property:font

<img width="332" alt="Image" src="https://github.com/user-attachments/assets/2e668af2-9fde-4fa1-8761-64df333bf590" />

**Comment by jeddy3** [comment]
> The position/order should also be taken into account. I think we should split this up so that non-shorthand checks use a much simpler function. font-face: "Latin Modern Math", math isn't a shorthand and doesn't require complex parsing to list the font family names.

Sounds like a sensible approach if we stick with our own scanning in `findFontFamily`.

> I wonder if we can use csstree here?

I think it's worth exploring further.

@otomad The issue is labelled as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

You can either explore using csstree for this or splitting the checks and improving the scanning as described in https://github.com/stylelint/stylelint/issues/8436#issuecomment-2700061077.

**Comment by skyclouds2001** [comment]
I've also fired https://github.com/mdn/data/pull/945 to update this syntax for mdn-data
